Source/Core/Common/MemArenaWin.cpp

```diff
@@ -455,35 +459,170 @@ void* LazyMemoryRegion::Create(size_t size)
   if (size == 0)
     return nullptr;
 
-  void* memory = VirtualAlloc(nullptr, size, MEM_RESERVE | MEM_COMMIT, PAGE_READWRITE);
+  if (!m_memory_functions.m_api_ms_win_core_memory_l1_1_6_handle.IsOpen())
+    return nullptr;
+
+  // reserve block of memory
+  const size_t memory_size = Common::AlignUp(size, BLOCK_SIZE);
+  const size_t block_count = memory_size / BLOCK_SIZE;
+  u8* memory =
+      static_cast<u8*>(static_cast<PVirtualAlloc2>(m_memory_functions.m_address_VirtualAlloc2)(
+          nullptr, nullptr, memory_size, MEM_RESERVE | MEM_RESERVE_PLACEHOLDER, PAGE_NOACCESS,
+          nullptr, 0));
   if (!memory)
   {
-    NOTICE_LOG_FMT(MEMMAP, "Memory allocation of {} bytes failed.", size);
+    NOTICE_LOG_FMT(MEMMAP, "Memory reservation of {} bytes failed.", size);
     return nullptr;
   }
 
+  // split into individual block-sized regions
+  for (size_t i = 0; i < block_count - 1; ++i)
```

Comment:
What is the purpose of the `- 1` here? Is it wrong to call `VirtualFree` on the last block?

Reply:
Yeah, this is somewhat unintuitive. Calling `VirtualFree` with `MEM_RELEASE | MEM_PRESERVE_PLACEHOLDER` splits out the given `[address, address+size)` from a larger placeholder into its own; this results in up to three separate placeholders: The preceeding original placeholder up to but excluding `address`, the `[address, address+size)` placeholder, and the following `address+size` until the end of the original placeholder. As such, the last iteration is not necessary since the last placeholder is already implicitly created in the previous split.